src/k8s-engine/unixlikeIntegrations.ts

```diff
@@ -0,0 +1,141 @@
+   */
+  protected pathConflictManager = new PathConflictManager();
+
+  /*
+   * We want to watch /usr/local/bin for changes, but fs.watch won't watch it until it exists.
+   * Once there's a watcher on it, the directory can be deleted, and when it's recreated the
+   * original watcher will still work on it.
+   */
+  #binWatcher: fs.FSWatcher|null = null;
+  /*
+   * If there's a successful change in setIntegration, no need to act on the next integration
+   */
+  #ignoreNextChange = false;
+
+  constructor() {
+    this.setupBinWatcher();
+  }
+
+  async listIntegrations(): Promise<Record<string, boolean | string>> {
+    for (const name of Integrations) {
+      const linkPath = path.join('/usr/local/bin', name);
+      const desiredPath = resources.executable(name);
+
+      try {
+        const currentDest = await fs.promises.readlink(linkPath);
+
+        if (currentDest === desiredPath) {
+          this.#results[linkPath] = true;
+        } else {
+          this.#results[linkPath] = `Already linked to ${ currentDest }`;
+        }
+      } catch (error) {
+        if (error.code === 'ENOENT') {
+          this.#results[linkPath] = false;
+        } else if (error.code === 'EINVAL') {
+          this.#results[linkPath] = `File exists and is not a symbolic link`;
+        } else {
+          this.#results[linkPath] = `Can't link to ${ linkPath }: ${ error }`;
+        }
+      }
+    }
+
+    return this.#results;
+  }
+
+  async setIntegration(linkPath: string, state: boolean): Promise<string | undefined> {
+    const desiredPath = resources.executable(path.basename(linkPath));
+
+    this.#results[linkPath] = '';
+    if (state) {
+      try {
+        await fs.promises.symlink(desiredPath, linkPath, 'file');
+      } catch (err) {
+        const message = `Error creating symlink for ${ linkPath }:`;
+
+        console.error(message, err);
+        this.#results[linkPath] = `${ message } ${ err.message }`;
+
+        return this.#results[linkPath] as string;
+      }
+    } else {
+      try {
+        await fs.promises.unlink(linkPath);
+      } catch (err) {
+        const message = `Error unlinking symlink for ${ linkPath }`;
+
+        console.error(message, err);
+        this.#results[linkPath] = `${ message } ${ err.message }`;
+
+        return this.#results[linkPath] as string;
+      }
+    }
+    this.#ignoreNextChange = true;
+  }
+
+  listIntegrationWarnings(): void {
+    for (const name of Integrations) {
+      this.pathConflictManager.reportConflicts(name);
+    }
+  }
+
+  protected async testUsrLocalBin() {
+    try {
+      await fs.promises.access('/usr/local/bin', fs.constants.W_OK | fs.constants.X_OK);
+      this.#results['/usr/local/bin'] = '';
+      if (!this.#binWatcher) {
+        this.#binWatcher = fs.watch('/usr/local/bin', async(eventType, filename) => {
+          if (Integrations.includes(filename)) {
+            if (this.#ignoreNextChange) {
+              this.#ignoreNextChange = false;
```

Comment:
There seems to be a race here: `#ignoreNextChange` could be set for `kubectl`, and then we can get a notification for `helm` instead? Is it _that_ bad if we just emit a integration update for something we just set?

Reply:
I did this because of the duplicated notifications. I was aware of a possible race condition, but was never able to hit it, since my typing, even pressing return, isn't fast enough. The argument for pulling out this code is that users won't notice duplicate notifications, but if they do hit the race condition, a checkbox will be out of sync with its underlying link file existence, and that's bad.

Reply:
Ah, I didn't think there would be duplicates (thought the new one would just overwrite the old one); in that case, this is fine for now.